pkg/apis/pipeline/v1beta1/pipelinerun_types.go

```diff
@@ -98,10 +98,9 @@ func (pr *PipelineRun) IsCancelled() bool {
 	return pr.Spec.Status == PipelineRunSpecStatusCancelled
 }
 
-// GetRunKey return the pipelinerun key for timeout handler map
-func (pr *PipelineRun) GetRunKey() string {
-	// The address of the pointer is a threadsafe unique identifier for the pipelinerun
-	return fmt.Sprintf("%s/%p", pipeline.PipelineRunControllerName, pr)
+// GetNamespacedName returns a k8s namespaced name that identifies this PipelineRun
+func (pr *PipelineRun) GetNamespacedName() types.NamespacedName {
+	return types.NamespacedName{Namespace: pr.Namespace, Name: pr.Name}
```

Comment:
Should we include the `kind` in there too ? We can't have 2 pipelinerun with the same name, but nothing prevent us from having a pipelinerun and a taskrun with the same name (which could be a problem then, as the key would be similar)

Reply:
we could - in the current implementation we have a timeout handler per controller, i.e. one for taskruns and one for pipelineruns. maybe it's better to be safe, but if we go this route we can use the existing NamespacedName type - maybe im just lazy :D

Reply:
@bobcatfish oh I see, right, then I think it's ok for now :wink: